README.md

```diff
@@ -114,43 +161,272 @@ encoding as JSON. Since Python is a dynamic language, when decoding via
+
+    # Run the worker until the event is set
+    worker = Worker(client, task_queue="my-task-queue", workflows=[MyWorkflow], activities=[my_activity])
+    async with worker:
+        await stop_event.wait()
+```
+
+Some things to note about the above code:
+
+* This creates/uses the same client that is used for starting workflows
+* While this example accepts a stop event and uses `async with`, `run()` and `shutdown()` may be used instead
+* Workers can have many more options not shown here (e.g. data converters and interceptors)
+
+### Workflows
+
+#### Definition
+
+Workflows are defined as classes decorated with `@workflow.defn`. The method invoked for the workflow is decorated with
+`@workflow.run`. Methods for signals and queries are decorated with `@workflow.signal` and `@workflow.query`
+respectively. Here's an example of a workflow:
+
+```python
+import asyncio
+from dataclasses import dataclass
+from datetime import timedelta
+from temporalio import activity, workflow
+from temporalio.client import Client
+from temporalio.worker import Worker
+
+@dataclass
+class GreetingInfo:
+    salutation: str = "Hello"
+    name: str = "<unknown>"
+
+@workflow.defn
+class GreetingWorkflow:
+    def __init__() -> None:
+        self._current_greeting = "<unset>"
+        self._greeting_info = GreetingInfo()
+        self._greeting_info_update = asyncio.Event()
+        self._complete = asyncio.Event()
+
+    @workflow.run
+    async def run(self, name: str) -> str:
+        self._greeting_info.name = name
+        while True:
+            # Store greeting
+            self._current_greeting = await workflow.execute_activity(
+                create_greeting_activity,
+                self._greeting_info,
+                start_to_close_timeout=timedelta(seconds=5),
+            )
+            workflow.logger.debug("Greeting set to %s", self._current_greeting)
+            
+            # Wait for salutation update or complete signal (this can be
+            # cancelled)
+            await asyncio.wait(
+                [self._greeting_info_update.wait(), self._complete.wait()],
+                return_when=asyncio.FIRST_COMPLETED,
+            )
+            if self._greeting_info_update.is_set():
+                self._greeting_info_update.clear()
+            elif self._complete.is_set():
+                return self._current_greeting
+
+    @workflow.signal
+    async def update_salutation(self, salutation: str) -> None:
+        self._greeting_info.salutation = salutation
+        self._greeting_info_update.set()
+
+    @workflow.signal
+    async def complete_with_greeting(self) -> None:
+        self._complete.set()
+
+    @workflow.query
+    async def current_greeting(self) -> str:
+        return self._current_greeting
 
 @activity.defn
-async def say_hello_activity(name: str) -> str:
-    return f"Hello, {name}!"
+async def create_greeting_activity(info: GreetingInfo) -> str:
```

Comment:
nit: I'd call this `greet_activity`, `create` makes it sound like you're creating an activity.

Reply:
It's a test, and I am creating a greeting not greeting someone.